### DISCUSSION TITLE
Message definition support for ROS2 .db3 recordings

### DISCUSSION THREAD

**Discussion by alecive** [foxglove/community#904] [Q&A]
I am a first time user of foxglove. I have a quick question: now that message definitions are supported in the sqlite3 storage plugin [LINK](https://github.com/ros2/rosbag2/pull/1293), is foxglove able to leverage this? 

Thanks!

**Comment by defunctzombie** [comment]
I don't recall that we ever added support for message definitions in sqlite3 files to Foxglove since that was added. We support MCAP which is the newer format for recording in ROS and do read message definitions from MCAP files.

Adding support for reading the message definitions from the sqlite3 files should be doable but is not on our radar. If you open a PR for it we would review it though.

**Reply by alecive** [reply]
Ok sounds good! Thanks.
